.github/workflows/wheels.yml

```diff
@@ -239,10 +239,7 @@ jobs:
         uses: actions/setup-python@v4
 
       - name: Upload artifacts
-        env:
-          # Secret variables need to be mapped to environment variables explicitly
-          SCIKIT_LEARN_NIGHTLY_UPLOAD_TOKEN: ${{ secrets.SCIKIT_LEARN_NIGHTLY_UPLOAD_TOKEN }}
-          SCIKIT_LEARN_STAGING_UPLOAD_TOKEN: ${{ secrets.SCIKIT_LEARN_STAGING_UPLOAD_TOKEN }}
-          ARTIFACTS_PATH: dist/artifact
-        # Force a replacement if the remote file already exists
-        run: bash build_tools/github/upload_anaconda.sh
+        uses: scientific-python/upload-nightly-action@b67d7fcc0396e1128a474d1ab2b48aa94680f9fc # 0.5.0
+        with:
+          artifacts_path: dist/artifact
+          anaconda_nightly_upload_token: ${{ secrets.SCIKIT_LEARN_NIGHTLY_UPLOAD_TOKEN }}
```

Comment:
This GitHub action only uploads scikit-learn nightly builds to the [scientific-python-nightly-wheels channel](https://anaconda.org/scientific-python-nightly-wheels).

However our existing `build_tools/github/upload_anaconda.sh` script also handles the [staging of wheels](https://anaconda.org/scikit-learn-wheels-staging/) generated from a stable release branch as part of our [release automation workflow](https://github.com/scikit-learn/scikit-learn/blob/main/.github/workflows/publish_pypi.yml#L28-L37) for pypi uploads.


Is there a way to configure the github action to also handle staging to a project specific channel?

Otherwise we would need a condition to either use our existing (simplified) script for staging and the shared github action for nightly builds.

Reply:
I saw that the script also handles wheels staging, that's why I did not remove it from the workflow. I will work on staging github action

Reply:
> that's why I did not remove it from the workflow.

I don't understand this sentence: this PR (#28560) deletes the line 248 so the script `build_tools/github/upload_anaconda.sh` is no longer called for build staging, right?

We cannot merge the PR as is. Otherwise the release process is no longer automated.

Reply:
I am working on the build staging part, in the incoming commits this script can be deleted. Perhaps I convert this PR to draft to avoid any confusion.